x/gov/keeper/tally.go

```diff
@@ -86,6 +87,48 @@ func (keeper Keeper) Tally(ctx sdk.Context, proposal v1.Proposal) (passes bool,
 	// If there is not enough quorum of votes, the proposal fails
 	percentVoting := totalVotingPower.Quo(sdk.NewDecFromInt(totalBondedTokens))
 	quorum, _ := sdk.NewDecFromStr(params.Quorum)
+	threshold, _ := sdk.NewDecFromStr(params.Threshold)
+
+	// Check if a proposal message is an ExecLegacyContent message
+	if len(proposal.Messages) > 0 {
+		var sdkMsg sdk.Msg
+		for _, msg := range proposal.Messages {
```

Comment:
Is there any way to ensure this kind of proposals (constitutional amendment or law) contains only one message ?

Because if I can submit a proposal that contains 2 `MsgExecLegacyContent`, where the first is a constitution amendment and the second a law proposal, the threshold and the quorum will be set to the ones of the laws and not the constituion amendment. It's not a problem when they are equal, but if not it will become a bug.

Reply:
see ValidateBasic. In principle it should, this is just to be sure it's compatible. Also the check on the number of messages is due to how tests were done (the tally bundles the proposal and a bank send) and in reality a proposal in v1 from what I can tell can be an arbitrary number of messages.

Reply:
the threshold is always the highest and law can be lower than regular and constitution can be lower then law. If in the prop there is a constitution amendment then that's what should rule above anything

Reply:
oh well yes I forgot about the check "greater than" in the code. So far so good.